cg/meta/workflow/mutant/metrics_parser/models.py

```diff
@@ -0,0 +1,33 @@
+from typing import Annotated
+from pydantic import BaseModel, BeforeValidator, ValidationError
+
+
+def empty_str_to_none(v: str) -> str | None:
+    return v or None
+
+
+def str_to_bool(v: str) -> bool:
+    if v == "TRUE":
+        return True
+    elif v == "FALSE":
+        return False
+    else:
+        raise ValidationError
+
+
+class SampleResults(BaseModel):
+    sample_name: str
+    selection: str
+    region_code: str
+    ticket: Annotated[int | None, BeforeValidator(empty_str_to_none)]
+    pct_n_bases: Annotated[float | None, BeforeValidator(empty_str_to_none)]
+    pct_10x_coverage: Annotated[float | None, BeforeValidator(empty_str_to_none)]
+    qc_pass: Annotated[bool | None, BeforeValidator(str_to_bool)]
+    lineage: str
+    pangolin_data_version: str
+    voc: str
+    mutations: Annotated[str | None, BeforeValidator(empty_str_to_none)]
```

Comment:
The default value will be set by the BeforeValidator, no?

Reply:
Objects of this model are created when the results file is parsed and, since the results file has to have this structure, there should always be a provided values, right? So the BeforeValidator will set it to None if the provided value is an empty string.

I'm thinking that None is easier to handle if, in the future, this information needs to be checked. Do you not agree?

Reply:
Decided to remove the BeforeValidators so this does not apply anymore.